File: projectforge-caldav/src/main/java/org/projectforge/caldav/cache/UserCache.java
package org.projectforge.caldav.cache;

import java.util.Date;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

import org.projectforge.caldav.model.AddressBook;
import org.projectforge.caldav.model.CalendarsHome;
import org.projectforge.caldav.model.ContactsHome;
import org.projectforge.caldav.model.User;
import org.springframework.stereotype.Component;

/**
 * Created by blumenstein on 06.11.16.
 */
@Component
public class UserCache
{
  private static long FIVE_MINUTES = 300000;

  private Map<Long, CalendarsHome> userCalendarHomeMap = new HashMap<>();

  private Map<Long, ContactsHome> userContactsHomeMap = new HashMap<>();

  private Map<Long, AddressBook> userAddressBookMap = new HashMap<>();

  private Map<User, Date> authorizedUserMap = new HashMap<>();

  public Map<User, Date> getAuthorizedUserMap()
  {
    return authorizedUserMap;
  }

  public boolean isUserAuthenticationValid(User user)
  {
    boolean userIsInList = false;
    Iterator<Map.Entry<User, Date>> it = authorizedUserMap.entrySet().iterator();
    while (it.hasNext()) {
      Map.Entry<User, Date> item = it.next();
      if (item.getKey().getPk().equals(user.getPk())) {
        Date now = new Date();
        if (now.getTime() - item.getValue().getTime() < FIVE_MINUTES) {
          userIsInList = true;
        } else {
          it.remove();
        }
      }
    }
    return userIsInList;
  }

  public Map<Long, CalendarsHome> getUserCalendarHomeMap()
  {
    return userCalendarHomeMap;
  }

  public Map<Long, ContactsHome> getUserContactsHomeMap()
  {
    return userContactsHomeMap;
  }

  public Map<Long, AddressBook> getUserAddressBookMap()
  {
    return userAddressBookMap;
  }
}

File: projectforge-caldav/src/main/kotlin/org/projectforge/caldav/cache/UserCache.kt
/////////////////////////////////////////////////////////////////////////////
//
// Project ProjectForge Community Edition
//         www.projectforge.org
//
// Copyright (C) 2001-2020 Micromata GmbH, Germany (www.micromata.com)
//
// ProjectForge is dual-licensed.
//
// This community edition is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License as published
// by the Free Software Foundation; version 3 of the License.
//
// This community edition is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
// Public License for more details.
//
// You should have received a copy of the GNU General Public License along
// with this program; if not, see http://www.gnu.org/licenses/.
//
/////////////////////////////////////////////////////////////////////////////

package org.projectforge.caldav.cache

import org.springframework.stereotype.Component

/**
 * Created by blumenstein on 06.11.16.
 */
@Component
class UserCache {

 /*   private val authorizedUserMap = mutableMapOf<User, Date>()

    fun getAuthorizedUserMap(): Map<User, Date> {
        return authorizedUserMap
    }

    fun isUserAuthenticationValid(user: User): Boolean {
        var userIsInList = false
        val it: MutableIterator<Map.Entry<User, Date>> = authorizedUserMap.entries.iterator()
        while (it.hasNext()) {
            val item = it.next()
            if (item.key.pk == user.pk) {
                val now = Date()
                if (now.time - item.value.time < FIVE_MINUTES) {
                    userIsInList = true
                } else {
                    it.remove()
                }
            }
        }
        return userIsInList
    }*/
}
